GitUI/NBugReports/BugReportInvoker.cs

```diff
@@ -101,6 +107,17 @@ public static void Report(Exception exception, bool isTerminating)
                 LogError(exception, isTerminating);
             }
 
+            if (exception is FileNotFoundException fileNotFoundException
+                && ReportFailToLoadAnAssembly(fileNotFoundException) == UserAction.RestartApplication)
+            {
```

Comment:
Here lies the problem. If you selects "Report issue" the app isn't getting closed, instead there's another dialog pops up. This needs fixing.

Reply:
I don't reproduce this behavior but I imagine that it depends where the exception is thrown.
Anyway, I have pushed a change so that the application exit (default behavior) after showing NBug window.

Reply:
> I don't reproduce this behavior

There are two conditions here:
1. `exception is FileNotFoundException`, and 
2. `& ReportFailToLoadAnAssembly(fileNotFoundException) == UserAction.RestartApplication`

So, if the first is true, the second may not be - if the user chooses another option - i.e., `ReportIssue`. In that case, the original implementation would fall through and show other popups...

We can't combine the both conditions, and we need to restructure it as follows:
```suggestion
            if (exception is FileNotFoundException fileNotFoundException)
            {
                switch (ReportFailToLoadAnAssembly(fileNotFoundException) 
                {
                    case UserAction.RestartApplication: { ... }
                    default: { ... }
                }
                
                return;
            }
```